reference/src/client/test/sa_key_provision_widevine.cpp

```diff
@@ -0,0 +1,348 @@
+        std::string rsa_key_type_string = "RSA-2048";
+        std::string rsa_key_container;
+       sa_status status = create_key_container(
+             rsa_key_type_string, //key type string
+             rsa_key_type, //key type
+             clear_rsa_key,
+             rsa_key_container);
+        ASSERT_EQ(SA_STATUS_OK, status);
+        INFO("rsa_key_container length : %d", rsa_key_container.size());
+
+        WidevineOemProvisioning *wvProvision = new WidevineOemProvisioning;
+        ASSERT_NE(nullptr, wvProvision);
+        wvProvision->oemDevicePrivateKey = (void*)rsa_key_container.data();
+        wvProvision->oemDevicePrivateKeyLength = rsa_key_container.size();
+
+        //Note: here just provide an examlpe to create certification data and passed
+        //into sa_key_provision_ta, how to use certification data is all up to
+        //SOC vendors.
+        auto certificate = random(CERTIFICATION_LENGTH);
+        wvProvision->oemDeviceCertificate = certificate.data();
+        wvProvision->oemDeviceCertificateLength = CERTIFICATION_LENGTH;
+
+        status = sa_key_provision_ta(WIDEVINE_OEM_PROVISIONING, wvProvision,0, nullptr);
+        if (nullptr != wvProvision) {
+           delete wvProvision;
+           wvProvision = nullptr;
+        }
+        ERROR("status : %d", status);
+        ASSERT_NE(status, SA_STATUS_OK);
+    }
+#ifdef FILE_BASED_FETCH_KEY
+    //There are two file widevine_oem_private.key and widevine_oem_cert.bin under
+    //tasecureapi/reference/src/client/,
+    //these files are widevine oem private key and certification. you can do
+    //"export  wievine_oem_privatekey=
+    //~/PATH/tasecureapi/reference/src/client/widevine_oem_private.key",
+    //"export  widevine_oem_cert=~/PATH/tasecureapi/reference/src/client/widevine_oem_cert.bin",
+    //the following test fromFileBased will pick up them and test
+    //Or
+    //you just simply copy these two files and put under /opt/drm/
+ TEST_F(SaKeyProvisionWidevineTest, fromFileBased) {
+        WidevineOemProvisioning *wvProvision = new WidevineOemProvisioning;
+        ASSERT_NE(nullptr, wvProvision);
+        ASSERT_TRUE(readWidevineData(&wvProvision));
+
+        sa_status status = sa_key_provision_ta(WIDEVINE_OEM_PROVISIONING, wvProvision,
+           sizeof(wvProvision), nullptr);
+        ASSERT_EQ(status, SA_STATUS_OK);
+
+        if (nullptr != wvProvision->oemDevicePrivateKey){
+           free(wvProvision->oemDevicePrivateKey);
+           wvProvision->oemDevicePrivateKey = nullptr;
+        }
+        if (nullptr != wvProvision->oemDeviceCertificate){
+           free(wvProvision->oemDeviceCertificate);
+           wvProvision->oemDeviceCertificate = nullptr;
+        }
+    }
+#endif //FILE_BASED_FETCH_KEY
+} // namespace
+
+#ifdef FILE_BASED_FETCH_KEY
+#include <sys/stat.h>
+#include <string.h>
+
+#define wievine_oem_privatekey "/opt/drm/widevine_oem_private.key"
+#define widevine_oem_cert      "/opt/drm/widevine_oem_cert.bin"
```

Comment:
also, we shouldn't use the /opt/drm folder as it is an implementation specific location.